### PR TITLE
Redesign

### PR DESCRIPTION
**Pull request by aminabukoash**

### CODE REVIEW

File: inc/fusion_algorithm.hpp [hunk 1]
```diff
@@ -1,9 +1,22 @@
-/** @file fusion_algorithm.h
+/** @file fusion_algorithm.hpp
```
Comment (aminabukoash) [inline inc/fusion_algorithm.hpp:2]: add file info

File: inc/sf_common.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/*
```
Comment (aminabukoash) [inline inc/sf_common.hpp]: add file info

File: inc/sf_impl.hpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/** @file sf_impl.hpp
+ * Contains the declarations for the sensor fusion implementation
+ */
+
```
Comment (aminabukoash) [inline inc/sf_impl.hpp:5]: add comments to all functions in file

File: inc/sf_common.hpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ * SfCommon.hpp
+ *
+ */
+
```
Comment (aminabukoash) [inline inc/sf_common.hpp:5]: add comments to all functions, strucs and lists in file

File: inc/sf_common.hpp [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+// Vector list of sensors
+typedef std::vector<Sensor_t> SensorsList_t;
+
+
+// Vector list of SensorList_t, for each SensorList_t (index) we will perform the fusion algorithm separately
+typedef std::vector<SensorsList_t> FusionList_t;
+
+//map to validate if sensor is stuck or not
+typedef std::map<string, SensorStuckInfo_t> StuckOrNotList_t;
```
Comment (aminabukoash) [inline inc/sf_common.hpp]: rename

File: src/sf_common.cpp [hunk 6]
```diff
@@ -0,0 +1,106 @@
+// All C++ Headers
```
Comment (aminabukoash) [inline src/sf_common.cpp:6]: add file info

File: src/sf_impl.cpp [hunk 7]
```diff
@@ -0,0 +1,248 @@
+/*
+ * sf_impl.cpp
```
Comment (aminabukoash) [inline src/sf_impl.cpp]: add file info

File: src/sf_impl.cpp [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+        printf("Either the filename is incorrect or it does not exists.\n");
+        return FAIL;
+    }
+    else
+    {
+        char line[1024];
+        int count = 0;
+        SensorsList_t sensorList;
+        FusionList_t fusionList;
+        StuckOrNotList_t sutckornotlist;
```
Comment (aminabukoash) [inline src/sf_impl.cpp]: rename maybe?

File: src/sf_impl.cpp [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+
+                    if (it != sutckornotlist.end())
+                    {
+                        printf("sensor: %s found with value %f\n",
+                               sensor.name,
+                               sensor.value);
+
+                        it->second.data.push_back(make_pair(sensor.time,
+                                                            sensor.value));
+//                        char time_buffer[100];
```
Comment (aminabukoash) [inline src/sf_impl.cpp]: remove

File: src/sf_impl.cpp [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+                    // handle sensor fusion lists
+                    int compare_status =
+                            compare_sensors_times(&sensorList[0],
+                                                  &sensor,
+                                                  fusion_interval_minutes);
+
+                    if (APPEND == compare_status)
+                    {
+                        sensorList.push_back(sensor);
+                        //TODO: should we check for duplicates?
```
Comment (aminabukoash) [inline src/sf_impl.cpp]: remove

File: src/sf_impl.cpp [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+                               it->second.data[i].second);
+
+                        it->second.status = SensorStatus_t::SENSOR_STATUS_STUCK;
+                    }
+                }
+            }
+            else
+            {
+                it->second.status = SensorStatus_t::SENSOR_STATUS_ON;
+
```
Comment (aminabukoash) [inline src/sf_impl.cpp:214]: delete line

File: src/source_fusion.cpp [hunk 12]
```diff
@@ -0,0 +1,85 @@
+
```
Comment (aminabukoash) [inline src/source_fusion.cpp]: add file info

File: src/source_fusion.cpp [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+int main(int argc, char *argv[])
+{
+    setbuf(stdout, NULL);
+    test_bench();
+    printf("Program Name Is: %s\n", argv[0]);
+    printf("Number of arguments: %d\n", argc);
+    //TODO: ask user to provide column data in file
```
Comment (aminabukoash) [inline src/source_fusion.cpp]: this is extra..

**Review by aminabukoash — COMMENTED** [review]
minor adjustments

**Comment by rishabhjiresal** [comment]
I think this branch looks outdated. Can you please make a pull request for the updated branch?

**Pull request closed without merging**